consensus/consensus.go

```diff
@@ -173,11 +174,26 @@ func (consensus *Consensus) GetBlockReward() *big.Int {
 	return consensus.lastBlockReward
 }
 
+// GetLeaderPrivateKey returns leader private key if node is the leader
+func (consensus *Consensus) GetLeaderPrivateKey(leaderKey *bls.PublicKey) (*bls.SecretKey, error) {
+	for i, key := range consensus.PubKey.PublicKey {
+		if key.IsEqual(leaderKey) {
+			return consensus.priKey.PrivateKey[i], nil
+		}
+	}
+	return nil, fmt.Errorf("leader private key not found")
```

Comment:
why this Errorf if not args, just make package level value of error proper , then wrap it with errors.Wrap with the extra info being which leader key was requested, being looked for